### ISSUE TITLE
opsgenie_schedule - There is already one Team with same name

### ISSUE THREAD

**Issue by bennylu2** [opsgenie/terraform-provider-opsgenie#353]
### Terraform Version
1.3.4

### Affected Resource(s)
- opsgenie_schedule

### Debug Output
```
data "opsgenie_team" "test" {
  name = "sre"
}

resource "opsgenie_schedule" "test" {
  name          = "sre"
  enabled       = false
  owner_team_id = data.opsgenie_team.test.id
}
```

```
│ Error: Error occurred with Status code: 409, Message: There is already one Team with same name, Took: 0.215000, RequestId: 3cb313f2-3c59-4bc3-a1ee-f1950e19a6ba
```

### Expected Behavior
Created schedule with defined name

### Actual Behavior
Error about duplicate **team name** when trying to create schedule which makes no sense. This happens during create and update operation. Appears you can't create schedule with same name as owner team name. This also happens in UI so appears to be a bug with the API

**Comment by diraol** [comment]
Yeah. the same apply for the "schedule", the tf provider is blocking from creating a schedule with the same name of a team.
